### ISSUE TITLE
Epic: Leveling System

### ISSUE THREAD

**Issue by ryanchartrand** [x-team/the-vault#52]
This Issue aims to complete the circle of the RPGified community. Its main goal is to help create a sense of progress for the coins system.

Earning gold coins will now be integrated into a leveling up system.

## **Below is a list of X-Team lifestyle Paths:**

**Maker** (Cooking, Building, Drones, Photography, OSS, etc.)
**Explorer** (Travel, X-Outpost, etc.)
**Seeker** (Conferences, Courses, Tutorials, etc.)
**Hero** (Exercise, Walking, Cycling, Yoga, etc.)
**Gamer** (Game Reviews, etc.)
**Strategist** (Strategies, Apps, Meditation, etc.)
**Unleasher** (Helping _others unleash_ through charity, parenting, mentoring, etc.)

A challenge can be created in any Club or Quest for any of these Paths.

After beating a challenge, you earn an Achievement for that Path. An Achievement also 'mines' 1 gold coin.

Beat 3 challenges in a Path and you can 'mine' 2 gold coins and Level Up to the next level.

**Examples:**
Join fit-quest and #cycling-club to level up your Hero path.
Join #cooking-club and level up your Maker path.
Join #commit-quest to find ways to level up all of your Paths.
Become a mentor to level up your Supporter path.
etc.

# **New Bot Commands**

## Award Achievement
*/award-achievement* [@username(s)] [category] "Achievement Text"

Ex: */award-achievement* @ryanchartrand productivity "The Strategist - Developed your own productivity strategy in Todoist."

Multi-user Ex: */award-achievement* @ryanchartrand @ozzy productivity "The Strategist - Developed your own productivity strategy in Todoist."

* This command should create an `achievement` for the user in the DB.
* If the user has reached 3 achievements for this Path, reset the achievement count to 0 and increase their Level for that Path by 1. Award the user 2 gold coins.
* Else -> Award only one gold coin to the user.

## View Achievements

*/achievements*

_Note: Screenshot is out of date and should reference Paths not Categories. It also shouldn't say "Welcome to Unleash" but instead "*Unleash Achievements*"

![image](https://user-images.githubusercontent.com/4460961/40201125-782ee8ea-5a1e-11e8-8265-6d55d7bb368e.png)

* Displays a list of all your Path Levels and # of achievements within each.
* Should utilize the colored vertical lines feature of Slack.
* Also need a Link below each Path to "View Achievements" if the user has achievements in that Path. It should then send a separate message with a list of Achievements in that Path.

------------------
Below this line is PHASE 2.

## View Channels with Challenges

*/unleash-challenges*

Displays a list of channels that offer challenges and for which categories.

## View Achievements per category

*unleash-achievements [category]*

Displays a list of all your achievements for a specific category.

## View Top 10 per category

*/unleash-leaderboard*

Displays a list of the Top 10 users for each category based on Levels.

**Comment by malgorzatajezierska** [comment]
This task has been divided into three epics and smaller tasks:
- The Vault misc: https://zube.io/x-team/internal-projects/epics/5
- The Vault Skac integration: https://zube.io/x-team/internal-projects/epics/3
- The Vault XHQ integration: https://zube.io/x-team/internal-projects/epics/4

**Comment by keremgocen** [comment]
is anyone working on this right now?

**Comment by malgorzatajezierska** [comment]
@keremgocen yes, @oktawianlagiewski is working on it now. This task remains as a master task, but it has been divided into smaller tasks in a new repository.
